crates/oxc_linter/src/rules/unicorn/escape_case.rs

```diff
@@ -88,20 +90,26 @@ fn check_case(value: &str, is_regex: bool) -> Option<String> {
                             p.next();
                             result.push('{');
                             for _ in 0..count {
-                                result.push(p.next().unwrap().to_ascii_uppercase());
+                                if let Some(nxt) = p.next() {
+                                    result.push(nxt.to_ascii_uppercase());
+                                }
                             }
                             p.next();
                             result.push('}');
                         }
                     } else if is_hex(&p, 4) {
                         for _ in 0..4 {
-                            result.push(p.next().unwrap().to_ascii_uppercase());
+                            if let Some(nxt) = p.next() {
+                                result.push(nxt.to_ascii_uppercase());
+                            }
                         }
                     }
                 }
                 'c' if is_regex => {
                     if matches!(p.clone().next(), Some(c) if c.is_ascii_lowercase()) {
-                        result.push(p.next().unwrap().to_ascii_uppercase());
+                        if let Some(nxt) = p.next() {
+                            result.push(nxt.to_ascii_uppercase());
+                        }
```

Comment:
this unwrap is ok as we check that `p.next()` is some above. but i think we can write it clearer as
```
if let Some(next) = p.clone().next() {
    if next.is_ascii_lowercase() {
        result.push...

Reply:
Good catch. That is it reverted.